ManageAxes.js

```diff
@@ -161,3 +171,82 @@ ManageAxes.resetAxes = async function()
+
+    // make a translation transform from the geometry to the world origin using the bounding box
+    const vec3d = await WSM.Geom.Vector3d(-(editingBBox.lower.x + editingBBox.upper.x) / 2, -(editingBBox.lower.y + editingBBox.upper.y) / 2, -editingBBox.lower.z);
+
+    // check if the transform from the geom to the world origin is null - if so, stop here
+    const isGeomTransf3dNull = await WSM.Vector3d.IsNull(vec3d)
+    if (isGeomTransf3dNull) {
+        // show a message that nothing was changed
+        await FormIt.UI.ShowNotification("The origin is already centered below this geometry.\nNo changes were made.", FormIt.NotificationType.Information, 0);
+
+        // no need to go any further
+        return;
+    }
+
+    // make a transform out of the vec3d
+    const geomTransf3d = await WSM.Transf3d.MakeTranslationTransform(vec3d);
+    // get the inverse transform for use later
+    const inverseGeomTransf3d = await WSM.Transf3d.Invert(geomTransf3d);
+
+    // if we got this far, changes will be made to the model, so start a new undo state
+    await FormIt.UndoManagement.BeginState();
+
+    // get all the non-owned objects in this history
+    const objectIds = await WSM.APIGetAllNonOwnedReadOnly(editingHistoryId);
+    // and move them to the world origin
+    await WSM.APITransformObjects(editingHistoryId, objectIds, geomTransf3d);
+
+    // get all the instances of this history
+    const instancesOfHistory = await WSM.APIGetAllAggregateTransf3dsReadOnly(editingHistoryId, 0);
+
+    // for each instance of this history...
+    for (let i = 0; i < instancesOfHistory.paths.length; i++) {
+        // how many levels "deep" is this instance from history 0
+        const historyDepthIndex = instancesOfHistory.paths[i].ids.length - 1;
+        const instanceObjectHistoryId = instancesOfHistory.paths[i].ids[historyDepthIndex];
+
+        // get the instance's transform and the inverse
+        const instanceTransf3d = instancesOfHistory.transforms[i];
+        const inverseInstanceTransf3d = await WSM.Transf3d.Invert(instanceTransf3d);
+
+        // multiply the geom inverse transform, the inverse instance transform, and the instance transform
+        const newTransf3dPartial = await WSM.Transf3d.Multiply(inverseGeomTransf3d, inverseInstanceTransf3d);
+        const newTransf3dFinal = await WSM.Transf3d.Multiply(instanceTransf3d, newTransf3dPartial);
+
+        // transform the instance by the final transf3d
+        await WSM.APITransformObject(instanceObjectHistoryId.History, instanceObjectHistoryId.Object, newTransf3dFinal);
+    }
+
+    await FormIt.UndoManagement.EndState("Manage Axes - Re-Origin");
+
+    // reset the local origin (this gets moved along with the non-owned objects) 
+    // so the local origin appears at the bottom center of the geometry
+    const defaultLCS = await WSM.Transf3d.Transf3d();
+    await WSM.APISetLocalCoordinateSystem(editingHistoryId, defaultLCS);
```

Comment:
Actually looking at this code more, I think it should just be removed. If the user explicitly made an LCS, the right thing to do is move it. If there is no explicit LCS, then there is nothing to do here. So in either case, this code is not required.

Reply:
This visibly moves the origin under the geometry, which is largely the point of this function. However, I did add a checkbox so if the customer doesn't want to override the existing origin, they can skip that part.